tools/setup/emoji/emoji_names.py

```diff
@@ -1577,4 +1577,1910 @@
+        "canonical_name": "man with white cane ",
+        "aliases": ["accessibility", "blind", "man", "man with white cane"],
+    },
+    "1f468-200d-1f9b0": {
+        "canonical_name": "man: red hair ",
+        "aliases": ["adult", "man", "red hair"],
+    },
+    "1f468-200d-1f9b1": {
+        "canonical_name": "man: curly hair ",
+        "aliases": ["adult", "curly hair", "man"],
+    },
+    "1f468-200d-1f9b2": {"canonical_name": "man: bald ", "aliases": ["adult", "bald", "man"]},
+    "1f468-200d-1f9b3": {
+        "canonical_name": "man: white hair ",
+        "aliases": ["adult", "man", "white hair"],
+    },
+    "1f468-200d-1f9bc": {
+        "canonical_name": "man in motorized wheelchair ",
+        "aliases": ["accessibility", "man", "man in motorized wheelchair", "wheelchair"],
+    },
+    "1f468-200d-1f9bd": {
+        "canonical_name": "man in manual wheelchair ",
+        "aliases": ["accessibility", "man", "man in manual wheelchair", "wheelchair"],
+    },
+    "1f468-200d-2695": {
+        "canonical_name": "man health worker ",
+        "aliases": ["doctor", "healthcare", "man", "man health worker", "nurse", "therapist"],
+    },
+    "1f468-200d-2696": {
+        "canonical_name": "man judge ",
+        "aliases": ["judge", "justice", "man", "scales"],
+    },
+    "1f468-200d-2708": {"canonical_name": "man pilot ", "aliases": ["man", "pilot", "plane"]},
+    "1f468-200d-2764-200d-1f468": {
+        "canonical_name": "couple with heart: man, man ",
+        "aliases": ["couple", "couple with heart", "love", "man"],
+    },
+    "1f468-200d-2764-200d-1f48b-200d-1f468": {
+        "canonical_name": "kiss: man, man ",
+        "aliases": ["couple", "kiss", "man"],
+    },
+    "1f469-200d-1f33e": {
+        "canonical_name": "woman farmer ",
+        "aliases": ["farmer", "gardener", "rancher", "woman"],
+    },
+    "1f469-200d-1f373": {"canonical_name": "woman cook ", "aliases": ["chef", "cook", "woman"]},
+    "1f469-200d-1f37c": {
+        "canonical_name": "woman feeding baby ",
+        "aliases": ["baby", "feeding", "nursing", "woman"],
+    },
+    "1f469-200d-1f393": {
+        "canonical_name": "woman student ",
+        "aliases": ["graduate", "student", "woman"],
+    },
+    "1f469-200d-1f3a4": {
+        "canonical_name": "woman singer ",
+        "aliases": ["actor", "entertainer", "rock", "singer", "star", "woman"],
+    },
+    "1f469-200d-1f3a8": {
+        "canonical_name": "woman artist ",
+        "aliases": ["artist", "palette", "woman"],
+    },
+    "1f469-200d-1f3eb": {
+        "canonical_name": "woman teacher ",
+        "aliases": ["instructor", "professor", "teacher", "woman"],
+    },
+    "1f469-200d-1f3ed": {
+        "canonical_name": "woman factory worker ",
+        "aliases": ["assembly", "factory", "industrial", "woman", "worker"],
+    },
+    "1f469-200d-1f466-200d-1f466": {
+        "canonical_name": "family: woman, boy, boy ",
+        "aliases": ["boy", "family", "woman"],
```

Comment:
In our system, a given name maps to a single emoji. A `canonical_name` or an alias for any given emoji becomes a name for that emoji, and can only point to one of them.

These terms that are populating `aliases` here aren't meant in CLDR to be aliases in that sense -- they're more like search keywords, where the same keyword can naturally be present on many items. Some discussion here: https://github.com/zulip/zulip/issues/18121#issuecomment-1110008908

Pending #18121, we'll want a story for what to do with CLDR's keywords when importing their emoji names. It should avoid creating duplicate aliases.

A very simple thing to do would be to drop the keywords in this CLDR import. But for many emoji, they provide useful synonyms or alternative cultural interpretations.

A still pretty simple rule that might get us a long way is to drop any keywords that also appear in the canonical name (in particular as whole words, with no letters before or after). If the user searches with those keywords in a way that would match the aliases, then it would also match the name directly.

(See https://github.com/zulip/zulip/commit/61b12dc25c4c260041774279111ad1b7a461209b / https://github.com/zulip/zulip/pull/21871 for recent changes to emoji search. It still won't get everything one might want -- in the examples above, I believe it'd currently work for "factory worker" but not for "boy boy" (tripped up by punctuation) or for "woman worker" or "worker factory" (won't match out of order.) But the aliases wouldn't help, because we don't try to spread a query across multiple names for an emoji anyway.)

Then after that… maybe just drop any aliases that are still duplicates. We could probably do something cleverer than that, but I think effort in that direction is probably best spent working toward making the search compatible with using CLDR-style keywords instead of aliases, as part of moving toward #18121.